src/utils/utils.ts

```diff
@@ -147,3 +147,33 @@ export const getGraphDetails = async () => {
     codapInterface.sendRequest({ action: "get", resource: `component[${g.id}]` }) as Promise<IResult>
   )).then(results => results.map(r => r.values));
 };
+
+export const getGraphByID = async (id: string) => {
+  const response = await codapInterface.sendRequest({ action: "get", resource: `component[${id}]` }) as IResult;
+  return response.values;
+};
+
+export const getDataContexts = async () => {
+  const contexts = await getListOfDataContexts();
+  const contextsDetails: Record<string, any> = {};
+  for (const ctx of contexts.values) {
+    const { name } = ctx;
+    const ctxDetails = await getDataContext(name);
+    contextsDetails[name] = ctxDetails.values;
+  }
+  return contextsDetails;
+};
+
+export const sendCODAPRequest = async (request: any) => {
+  const response = await codapInterface.sendRequest(request);
+  return response;
+};
```

Comment:
Looks like this new `sendCODAPRequest` function doesn't currently get called from anywhere. Should it be removed? Or is there maybe something that calls it that didn't get committed?